### ISSUE TITLE
Unresolved import error on struct vs unite-like struct

### ISSUE THREAD

**Issue by malbarbo** [rust-lang/rust#27695]
```rust
pub use m::A;

mod m {
    use super::*;
    pub struct A;
    //pub struct A {
    //    a: bool
    //}
}
```

The code compile with first definition of A, but not the second.

This may be related to #26873.

**Comment by mitaa** [comment]
This compiles on playpen nightly.

**Comment by steveklabnik** [comment]
Yes, it fails on stable and beta, but works on nightly. Therefore, I'm tempted to close it, as it's seemingly already fixed, but let's see if anyone else can weigh in on the exact details.

**Comment by alexcrichton** [comment]
I think this was either fixed in #27547 or #27439, so closing as working (yay!)
